tests/unit/amo/components/TestAddonReviewList.js

```diff
@@ -1,122 +1,143 @@
-import { shallow } from 'enzyme';
 import React from 'react';
 
 import fallbackIcon from 'amo/img/icons/default-64.png';
 import createStore from 'amo/store';
 import { fetchReviews, setAddonReviews } from 'amo/actions/reviews';
 import { setViewContext } from 'amo/actions/viewContext';
-import {
+import AddonReviewList, {
   AddonReviewListBase,
-  mapStateToProps,
 } from 'amo/components/AddonReviewList';
+import AddonReviewListItem from 'amo/components/AddonReviewListItem';
 import NotFound from 'amo/components/ErrorPage/NotFound';
 import Link from 'amo/components/Link';
 import Paginate from 'core/components/Paginate';
 import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+} from 'core/constants';
+import {
   fetchAddon, createInternalAddon, loadAddons,
 } from 'core/reducers/addons';
 import ErrorList from 'ui/components/ErrorList';
-import Rating from 'ui/components/Rating';
-import { fakeAddon, fakeReview, dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import { fakeAddon, fakeReview } from 'tests/unit/amo/helpers';
 import {
   createFetchAddonResult,
   createStubErrorHandler,
   getFakeI18nInst,
+  shallowUntilTarget,
 } from 'tests/unit/helpers';
 import { setError } from 'core/actions/errors';
 import { createApiError } from 'core/api/index';
 import LoadingText from 'ui/components/LoadingText';
 
-function getLoadedReviews({
-  addonSlug = fakeAddon.slug, reviews = [fakeReview], reviewCount = 1 } = {},
-) {
-  const action = setAddonReviews({ addonSlug, reviewCount, reviews });
-  // This is how reviews look after they have been loaded.
-  return action.payload.reviews;
-}
-
-describe('amo/components/AddonReviewList', () => {
-  describe('<AddonReviewListBase/>', () => {
-    function render({
-      addon = fakeAddon,
-      dispatch = sinon.stub(),
-      errorHandler = createStubErrorHandler(),
-      params = {
-        addonSlug: fakeAddon.slug,
-      },
-      reviews = [fakeReview],
-      ...customProps
-    } = {}) {
-      const loadedReviews = reviews ? getLoadedReviews({ reviews }) : null;
-      const props = {
-        addon,
-        dispatch,
-        errorHandler,
-        i18n: getFakeI18nInst(),
-        location: { query: {} },
-        params,
-        reviewCount: loadedReviews && loadedReviews.length,
-        reviews: loadedReviews,
-        ...customProps,
-      };
-
-      return shallow(<AddonReviewListBase {...props} />);
-    }
-
-    it('requires an addonSlug property', () => {
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = createStore().store;
+  });
+
+  const render = ({
+    params = {
+      addonSlug: fakeAddon.slug,
+    },
+    ...customProps
+  } = {}) => {
+    const props = {
+      i18n: getFakeI18nInst(),
+      location: { query: {} },
+      params,
+      store,
+      ...customProps,
+    };
+    return shallowUntilTarget(
```

Comment:
Believe it or not I didn't want to spend time converting this whole suite to shallow rendering. However, since I had to refactor the `reviews` state *and* be sure I didn't break things along the way, it was really helpful to switch to shallow rendering first. It turned out to be a nice validation of this `shallowUntilTarget` pattern because you can see how this component's reliance on mapped state is fully tested now.